specification.markdown

```diff
@@ -64,30 +64,78 @@ followed by the Data Size, and then the non-interpreted Binary itself:
-    will be sufficient for the time being.
-
-    There is only one reserved word for Element Size encoding, which is
-    an Element Size encoded to all 1's. Such a coding indicates that the
-    size of the Element is unknown, which is a special case that we
-    believe will be useful for live streaming purposes. However, avoid
-    using this reserved word unnecessarily, because it makes parsing
-    slower and more difficult to implement.
+## Element Data Size
+
+The Element Data Size expresses the length in octets of Element Data.
+The Element Data Size itself MUST be encoded as a Variable Size Integer.
+By default, EBML Element Data Sizes can be encoded in lengths from one
+octet to eight octets, although Element Data Sizes of greater lengths
+MAY be used if the octet length of the EBML document's longest Element
+Data Size is declared in the EBMLMaxSizeLength Element of the EBML
+Header. Unlike the VINT\_DATA of the Element ID, the VINT\_DATA
+component of the Element Data Size is NOT REQUIRED to be encoded at the
+shortest valid length. For example, an Element Data Size with binary
+encoding of 1011 1111 or a binary encoding of 0100 0000 0011 1111 are
+both valid Element Data Sizes and both store a semantically equal value.
+
+Although an Element ID with all VINT\_DATA bits set to zero is invalid,
+an Element Data Size with all VINT\_DATA bits set to zero is allowed
+for EBML Data Types which do not mandate a non-zero length. In
+particular the Signed Integer, Unsigned Integer, Float, and Date EBML
+Element Data Types have definitions which require a length of at least
+one octet and thus in these cases an Element Data Size with all
+VINT\_DATA bits set to zero is invalid. EBML Element Data Types such as
+String, UTF-8, Master-element, and Binary do not have definitions that
+include minimum lengths greater than one octet and thus Elements of
+these Data Types may include an Element Data Size with all VINT\_DATA
+bits set to zero. This indicates that the Element Data of the Element
+is zero octets in length. Such an Element is referred to as an Empty
+Element.
+
+An Element Data Size with all VINT\_DATA bits set to one is reserved as
+an indicator that the size of the Element is unknown. The only reserved
+value for the VINT\_DATA of Element Data Size is all bits set to one.
+This rule allows for an Element to be written and read before the size
+of the Element is known; however unknown Element Data Size values SHOULD
+NOT be used unnecessarily. An Element with an unknown Element Data Size
+MUST be a Master-element in that it contains other EBML Elements as
+sub-elements. The end of the Master-element is determined by the
+beginning of the next element that is not a valid sub-element of the
+Master-element. The use of Elements of unknown size is dependent on the
+definition of the EBML Schema declared in DocType, because an Element of
+unknown size can not be parsed without a complete list of all possible
+sub-elements.
```

Comment:
Again, it can be parsed with a partial list of all possible sub-elements and even a partial list of upper elements. Unknown EBML elements will just be skipped until something known (immedialy lower or higher) is found.
Without this EBML files would not be backward compatible.